kasa/smart/modules/ledmodule.py

```diff
@@ -0,0 +1,68 @@
+"""Module for led controls."""
+from typing import TYPE_CHECKING, Dict
+
+from ...feature import Feature, FeatureType
+from ..smartmodule import SmartModule
+
+if TYPE_CHECKING:
+    from ..smartdevice import SmartDevice
+
+
+class LedModule(SmartModule):
+    """Implementation of led controls."""
+
+    REQUIRED_COMPONENT = "led"
+    QUERY_GETTER_NAME = "get_led_info"
+
+    def __init__(self, device: "SmartDevice", module: str):
+        super().__init__(device, module)
+        self._add_feature(
+            Feature(
+                device=device,
+                container=self,
+                name="LED",
+                icon="mdi:led-{state}",
+                attribute_getter="led",
+                attribute_setter="set_led",
+                type=FeatureType.Switch,
+            )
+        )
+
+    def query(self) -> Dict:
+        """Query to execute during the update cycle."""
+        return {self.QUERY_GETTER_NAME: {"led_rule": None}}
+
+    @property
+    def mode(self):
+        """LED mode setting.
+
+        "always", "never", "night_mode"
+        """
+        return self.data["led_rule"]
+
+    @property
+    def led(self):
+        """Return current led status."""
+        return self.data["led_status"]
+
+    async def set_led(self, enable: bool):
+        """Set led.
+
+        This should probably be a select with always/never/nightmode.
```

Comment:
Maybe this should be a select from the outset as it may be awkward to change in HA afterwards otherwise

Reply:
That's a good point... The problem I'm seeing is that we are introducing quite a few new platforms if we want to get all done at once (binary_sensor, select, button) so I was hoping we could start with something small first.

I think we should still merge this already, and see how far we get before we want to call it enough for a new release.